src/C-DEngine/C-DEngines/C-DThingService/cdeTheThingRegistry.cs

```diff
@@ -488,9 +500,14 @@ public  static bool UnregisterThingGlobally(TheThing pThing)
         /// <returns></returns>
         internal static bool SyncGlobalThings(Guid nodeId, List<TheThing> things, bool OnlyMerge=false)
         {
-            if (TheCDEngines.MyThingEngine == null || TheCDEngines.MyThingEngine.MyThingRegistry == null || things == null ||
-                TheCDEngines.MyThingEngine.MyThingRegistry.MyThings == null)
+            if (things == null || nodeId == TheBaseAssets.MyServiceHostInfo?.MyDeviceInfo?.DeviceID || TheCDEngines.MyThingEngine?.MyThingRegistry?.MyThings == null)
+                return false;
+
+            if (things.Any((t) => t.cdeN != nodeId)) // Don't take things from nodes that are not the originator: assume malicious sender and reject entire list
+            {
+                TheBaseAssets.MySYSLOG.WriteToLog(13424, new TSM(eEngineName.ThingService, $"Received invalid global thing list for node {nodeId}. Rejecting list.", eMsgLevel.l1_Error, ""));
```

Comment:
How often could this happen? Shall we rather put this on a higher debug-level? (TSM.L(Essentials))?

Reply:
This would only happen if there's a malicious node sending bad TSMs. Probably want to know about that, but at the same time it's a log-spew "DoS".